### PR TITLE
Docs: getopt is deprecated in Python 3.13

### PR DESCRIPTION
**Pull request by hugovk**
Let's list it alongside `optparse` on https://docs.python.org/3.13/library/superseded.html.

Re:

* https://docs.python.org/3.13/whatsnew/3.13.html#deprecated
* https://github.com/python/cpython/issues/106535

----
:books: Documentation preview :books:: https://cpython-previews--109438.org.readthedocs.build/

### CODE REVIEW

**Comment by vstinner** [comment]
>  Docs: getopt is deprecated in Python 3.13 #109438 

It's only **soft** deprecated! That's very different from being deprecated ;-) Some people already got upset that I **soft** deprecated getopt! https://docs.python.org/dev/glossary.html#term-soft-deprecated

File: Doc/library/superseded.rst [hunk 1]
```diff
@@ -10,4 +10,5 @@ backwards compatibility. They have been superseded by other modules.
 
 .. toctree::
 
+   getopt.rst
```
Comment (vstinner) [inline Doc/library/superseded.rst:13]: Please update the introduction to say:

```
deprecated or :term:`soft deprecated`
```

Comment (hugovk) [reply Doc/library/superseded.rst:13]: Updated!

Comment (vstinner) [reply Doc/library/superseded.rst:13]: Thanks.

**Comment by vstinner** [comment]
I suggest fixing argparse in this case.

**Comment by ericvsmith** [comment]
As the long (13 year!) discussion on the noted ticket points out, this isn't possible without rewriting argparse. And it's also impossible without breaking existing code or adding a separate parsing algorithm. So I don't think that's a viable course of action. Which leaves us without a usable, un-deprecated option parser. If that's our position, I think we should also soft deprecate argparse. I'd be okay with that.

**Comment by ericvsmith** [comment]
I'm only commenting here because I think the soft deprecation of getopt was a mistake without looking at the larger picture.

I'll open a discussion on d.p.o when I have some time. Or maybe bring it up at the core sprint for starters.

**Pull request merged**